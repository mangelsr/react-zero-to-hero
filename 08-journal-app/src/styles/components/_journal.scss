.journal__main-content {
    display: flex;
}

.journal__sidebar {
    background-color: $dark-gray;
    color: white;
    display: flex;
    flex-direction: column;
    height: 100vh;
    padding: 0 10px;
    width: 450px;
}

.journal__sidebar-navbar {
    display: flex;
    justify-content: space-between;
}

.journal__sidebar-navbar h3 {
    font-weight: lighter;
}

.journal__new-entry {
    align-items: center;
    cursor: pointer;
    display: flex;
    justify-content: center;
    margin-top: 30px;
    width: 100%;
    flex-direction: column;

    transition: color .3s ease;

    &:hover {
        color: darken($color: white, $amount: 20);
    }
}

.journal__entries {
    flex: 1 1 auto;
    margin-top: 10px;
    overflow-y: auto;
}

.journal__entry {
    background-color: white;
    border-radius: 4px;
    color: $dark-gray;
    display: flex;
    margin-bottom: 10px;
    overflow: hidden;
    justify-content: space-between;
}

.journal__entry-picture {
    height: 75px;
    width: 75px;

}

.journall__entry-body {
    padding: 10px;
}

.journall__entry-title {
    font-size: 14px;
    font-weight: bold;
}

.journall__entry-content {
    font-size: 10px;
}

.journall__entry-date-box {
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding: 5px;
}

.journall__entry-date-box span {
    font-size: 12px;
}