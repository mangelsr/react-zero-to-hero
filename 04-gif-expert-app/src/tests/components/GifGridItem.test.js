import '@testing-library/jest-dom';
import React from 'react'
import { shallow } from 'enzyme';

import { GifGridItem } from '../../components/GifGridItem';


describe('Pruebas de <GifGridItem />', () => {

    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    
    test('Debe hacer match con el snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de tener un parrafo con el titulo', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('Debe tener el url y el alt', () => {
        const img = wrapper.find('img').props();
        expect(img.src).toBe(url);
        expect(img.alt).toBe(title);
    });

    test('Debe tener las clases de css ', () => {
        const div = wrapper.find('div').props();
        const classes = div.className.split(' ');
        expect(classes).toContain('card');
        expect(classes).toContain('animate__animated');
        expect(classes).toContain('animate__fadeIn');
    });

});