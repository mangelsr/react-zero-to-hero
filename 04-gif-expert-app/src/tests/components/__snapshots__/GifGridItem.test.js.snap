// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGridItem /> Debe hacer match con el snapshot 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
